Bug: catching exception in pytest doesn't work with AnyIO

**Describe the bug**  
If you use AnyIO to test that an exception is raised in a handler, the test always fails.

**How to reproduce**

Take the example from the documentation: https://dummy.faststream.airt.ai/latest/getting-started/subscription/test/#catching-exceptions but use **AnyIO** instead of **pytest-asyncio**:

```python
import pytest
from faststream import FastStream
from faststream.confluent import KafkaBroker, TestKafkaBroker
from pydantic import ValidationError

broker = KafkaBroker("localhost:9092")
app = FastStream(broker)


@pytest.fixture
def anyio_backend():
    return "asyncio"


@broker.subscriber("test-topic")
async def handle(
    name: str,
    user_id: int,
):
    assert name == "John"
    assert user_id == 1


@pytest.mark.anyio
async def test_validation_error():
    async with TestKafkaBroker(broker) as br:
        with pytest.raises(ValidationError) as ex_info:
            await br.publish("wrong message", topic="test-topic")
        assert ex_info.match("1 validation error for handle")
```

**Expected behavior**  
The test should pass successfully.

**Observed behavior**  
The test fails, indicating that the exception is not being handled:

```
================================================================================== short test summary info ===================================================================================
FAILED tests/test_error.py::test_validation_error - pydantic_core._pydantic_core.ValidationError: 1 validation error for handle
===================================================================================== 1 failed in 0.58s ======================================================================================
```

**Environment**

```
Running FastStream 0.5.48 with CPython 3.12.9 on Darwin
```

**Additional context**  
Inspecting the internals of FastStream, it looks like a `Future` is created in the current loop and the exception is set. It seems that AnyIO interprets this unhandled `Future` exception as an unhandled error in the test.

If you catch the future error with:

```python
        try:
            await handle.wait_call(1)
        except Exception: ...
```

The test works as expected.

So this example works:

```python
import pytest
from faststream import FastStream
from faststream.confluent import KafkaBroker, TestKafkaBroker
from pydantic import ValidationError

broker = KafkaBroker("localhost:9092")
app = FastStream(broker)


@pytest.fixture
def anyio_backend():
    return "asyncio"


@broker.subscriber("test-topic")
async def handle(
    name: str,
    user_id: int,
):
    assert name == "John"
    assert user_id == 1


@pytest.mark.anyio
async def test_validation_error():
    async with TestKafkaBroker(broker) as br:
        with pytest.raises(ValidationError) as ex_info:
            await br.publish("wrong message", topic="test-topic")
        assert ex_info.match("1 validation error for handle")
        try:
            await handle.wait_call(1)
        except Exception: ...
```

--- a/faststream/_internal/endpoint/call_wrapper.py
+++ b/faststream/_internal/endpoint/call_wrapper.py
@@ -138,6 +138,8 @@ class HandlerCallWrapper(Generic[P_HandlerParams, T_HandlerReturn]):
 
         if error:
             self.future.set_exception(error)
+            # Mark the mirrored error as retrieved to avoid unhandled-future reports.
+            self.future.exception()
 
         else:
             self.future.set_result(result)

--- a/tests/utils/test_handler_call_wrapper.py
+++ b/tests/utils/test_handler_call_wrapper.py
@@ -1,0 +1,39 @@
+import gc
+
+import pytest
+
+from faststream._internal.endpoint.call_wrapper import HandlerCallWrapper
+
+
+@pytest.mark.anyio()
+@pytest.mark.parametrize("anyio_backend", ("asyncio",))
+async def test_handler_exception_does_not_leak_to_event_loop(
+    anyio_backend: str,
+) -> None:
+    async def handler() -> None:
+        return None
+
+    wrapper = HandlerCallWrapper(handler)
+    wrapper.set_test()
+    error = ValueError("handler failed")
+
+    wrapper.trigger(error=error)
+    del wrapper
+    gc.collect()
+
+
+@pytest.mark.asyncio()
+async def test_handler_exception_remains_available_to_wait_call() -> None:
+    async def handler() -> None:
+        return None
+
+    wrapper = HandlerCallWrapper(handler)
+    wrapper.set_test()
+    error = ValueError("handler failed")
+
+    wrapper.trigger(error=error)
+
+    with pytest.raises(ValueError, match="handler failed") as exc_info:
+        await wrapper.wait_call()
+
+    assert exc_info.value is error

